packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart

```diff
@@ -102,23 +88,55 @@ void main() {
 
       when(subscribeInvocation)
           .thenAnswer((_) => notificationStreamController.stream);
+
+      when(notifyInvocation).thenAnswer(
+        (_) async {
+          final testIp = '123.123.123.123';
+          final portA = 10456;
+          final portB = 10789;
+
+          notificationStreamController.add(
+            AtNotification.empty()
+              ..id = Uuid().v4()
+              ..key = '$sessionId.${Sshrvd.namespace}'
+              ..from = '@sshrvd'
+              ..to = '@client'
+              ..epochMillis = DateTime.now().millisecondsSinceEpoch
+              ..value = '$testIp,$portA,$portB',
+          );
+        },
+      );
     }
 
     test('Initialization - sshrvd host', () async {
       /// Set the required parameters
       whenInitializationWithSshrvdHost();
-
       expect(stubbedSshrvdChannel.sshrvdAck, SshrvdAck.notAcknowledged);
-      expect(stubbedSshrvdChannel.initalizeStarted, false);
+      expect(stubbedSshrvdChannel.initializeStarted, false);
 
       verifyNever(subscribeInvocation);
       verifyNever(notifyInvocation);
 
-      await expectLater(stubbedSshrvdChannel.callInitialization(), completes);
+      await expectLater(stubbedSshrvdChannel.initialize(), completes);
 
       verifyInOrder([
-        subscribeInvocation,
-        notifyInvocation,
+        () => subscribeStub(
+            regex: '$sessionId.${Sshrvd.namespace}@', shouldDecrypt: true),
+        () => notifyStub(
+              any<AtKey>(
+                that: predicate(
+                  // Predicate matching specifically the sshrvdIdKey format
+                  (AtKey key) =>
+                      key.key == 'mydevice.${Sshrvd.namespace}' &&
+                      key.sharedBy == '@client' &&
+                      key.sharedWith == '@sshrvd' &&
+                      key.metadata != null &&
+                      key.metadata!.namespaceAware == false &&
+                      key.metadata!.ttl == 10000,
+                ),
+              ),
+              any(),
+            ),
```

Comment:
Here's the better argument matching as mentioned, checking that the values contained in the key passed in are as expected